src/array.cr

```diff
@@ -1585,6 +1620,17 @@ class Array(T)
     self
   end
 
+  # Swaps the elements at the given `index0` and `index1` and returns `self`.
+  # Raises an `IndexError` if an index given is out of bounds.
+  #
+  # ```
+  # a = ["first", "second", "third"]
+  # a.swap(1, 2)  #=> ["first", "third", "second"]
+  # a             #=> ["first", "third", "second"]
```

Comment:
Don't repeat the result

Reply:
No I think this is fine. We should demo that its mutating the array.

Reply:
Then just show the result of `a` only, to demonstrate the mutation

Reply:
IMO It's good to know that `Array#swap` returns the mutated array too.

Reply:
Would there be a good reason to have a non-mutating version `swap` and a mutating version `swap!` ?

I can't think of a good reason, other than consistency with the bang `!` vs non-bang methods.

Reply:
I don't think there's a good use case. One can always do `.dup.swap(...)` (most of Ruby methods just do this internally)

Reply:
Crystal methods as well (`#sort` for example).